### PR TITLE
feat: expose new container for 2-state navigators

### PR DESCRIPTION
**Pull request by WoLewicki**
## Description

PR adding `RNSScreenNavigationContainer` component on iOS. This component should be used in `bottom-tabs` and `drawer` navigators of `react-navigation` in order to resolve layout issues connected to integration between `UIViewController` and `UINavigationController` containers (see #619 and #564). `stack` navigator unfortunately cannot use `UINavigationController` since it needs the previous screens to be visible e.g. during the transition. Connected PR in `react-navigation`: https://github.com/react-navigation/react-navigation/pull/9772

## Changes

Added new component and `hasTwoStates` prop for `ScreenContainer` which controls which native component should be used.

## Test code and steps to reproduce

`Test619.js` and `Test564.js` to see the layout issues gone. It needs setting `hasTwoStates` props on `bottom-tabs` navigator in order for this to work.

## Checklist

- [x] Included code example that can be used to test this change
- [x] Updated TS types
- [x] Ensured that CI passes

### CODE REVIEW

File: src/types.tsx [hunk 1]
```diff
@@ -179,6 +179,10 @@ export interface ScreenContainerProps extends ViewProps {
    * A prop that gives users an option to switch between using Screens for the navigator (container). All children screens should have the same value of their "enabled" prop as their container.
    */
   enabled?: boolean;
+  /**
+   * A prop to be used in drawer and bottom-tabs navigators for better implementation of ScreenContainer on iOS.
+   */
+  tabsOrDrawer?: boolean;
```
Comment (satya164) [inline src/types.tsx:185]: I think a prop name like `hasTwoStates` (didn't think a lot about the name) or something like that would be more generic and more appropriate

File: src/index.native.tsx [hunk 2]
```diff
@@ -47,6 +47,8 @@ function enableScreens(shouldEnableScreens = true): void {
 
 // const that tells if the library should use new implementation, will be undefined for older versions
 const shouldUseActivityState = true;
+const isNavigationContainerAvailable =
```
Comment (satya164) [inline src/index.native.tsx:50]: Is there a case when this won't be available when someone is using this version of react-native-screens?

Comment (WoLewicki) [reply src/index.native.tsx:50]: What if someone updates to this version of library in Expo managed workflow? I know it should not be done, but 🤷‍♂️ .

Comment (satya164) [reply src/index.native.tsx:50]: @WoLewicki updating to an incompatible version of the library could also cause other issues as well, and surface bugs which are already fixed in latest version, so probably not a use case we should support.

**Review by kacperkapusciak — APPROVED** [review]

**Pull request merged**